builtin/logical/aws/backend.go

```diff
@@ -65,8 +66,17 @@ func Backend() *backend {
 		Invalidate:        b.invalidate,
 		WALRollback:       b.walRollback,
 		WALRollbackMinAge: minAwsUserRollbackAge,
-		PeriodicFunc:      b.rotateExpiredStaticCreds,
-		BackendType:       logical.TypeLogical,
+		PeriodicFunc: func(ctx context.Context, req *logical.Request) error {
+			repState := conf.System.ReplicationState()
+			if (conf.System.LocalMount() &&
+				!repState.HasState(consts.ReplicationPerformanceSecondary)) &&
+				!repState.HasState(consts.ReplicationDRSecondary) &&
+				!repState.HasState(consts.ReplicationPerformanceStandby) {
```

Comment:
I think this is evaluating to 

```
if (LocalVaultMount && PrimaryVaultNode)
```

But I think we want that to be an OR operation.